src/value-types.js

```diff
@@ -0,0 +1,218 @@
+var execute = require('./execute');
+
+var Value = function Value(value) {
+    this.value = value;
+};
+
+Value.prototype.equals = function(b) {
```

Comment:
I've never tried to write something using Parasitical OOP before. I'll do another PR with this to compare. What I think I'll miss with Parasitical OOP is that you can't use instanceof anymore .And instanceof that makes it easy to have `Function` & `InternalFunction` or checking that only `Value`s are pushed on the stack. later we could also have a `Number` extended by both `Integer` and `Float`, etc. 
anyway, let's try and see what it looks like